core/templates/domain/exploration/read-only-exploration-backend-api.service.ts

```diff
@@ -17,153 +17,187 @@
  * about explorations from the backend.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
-
-require(
-  'pages/exploration-player-page/exploration-player-page.constants.ajs.ts');
-
-angular.module('oppia').factory('ReadOnlyExplorationBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService',
-  'EXPLORATION_DATA_URL_TEMPLATE', 'EXPLORATION_VERSION_DATA_URL_TEMPLATE',
-  function(
-      $http, $q, UrlInterpolationService,
-      EXPLORATION_DATA_URL_TEMPLATE, EXPLORATION_VERSION_DATA_URL_TEMPLATE) {
-    // Maps previously loaded explorations to their IDs.
-    var _explorationCache = [];
-
-    var _fetchExploration = function(
-        explorationId, version, successCallback, errorCallback) {
-      var explorationDataUrl = _getExplorationUrl(explorationId, version);
-
-      $http.get(explorationDataUrl).then(function(response) {
-        var exploration = angular.copy(response.data);
-        if (successCallback) {
-          successCallback(exploration);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+import { AppConstants } from 'app.constants';
+import { ClassifierBackendDict } from 'domain/classifier/classifier.model';
+import { ParamChangeBackendDict } from 'domain/exploration/ParamChangeObjectFactory';
+import { ParamSpecsBackendDict } from 'domain/exploration/ParamSpecsObjectFactory';
+import { StateObjectsBackendDict } from 'domain/exploration/StatesObjectFactory';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+
+interface ReadOnlyExplorationBackendDict {
+  'init_state_name': string;
+  'param_changes': ParamChangeBackendDict[];
+  'param_specs': ParamSpecsBackendDict;
+  'states': StateObjectsBackendDict;
+  'title': string;
+  'language_code': string;
+  'objective': string;
+  'correctness_feedback_enabled': boolean;
+}
+
+export interface FetchExplorationBackendResponse {
+  'can_edit': boolean;
+  'exploration': ReadOnlyExplorationBackendDict;
+  'exploration_id': string;
+  'is_logged_in': boolean;
+  'session_id': string;
+  'version': number;
+  'preferred_audio_language_code': string;
+  'state_classifier_mapping': Record<string, ClassifierBackendDict>;
```

Comment:
Why is this used instead of StateClassifierMappingBackendDict?

Reply:
Record<string, type> acts like a dict with string keys and values of type `type`.